code/game/machinery/guestpass.dm

```diff
@@ -39,154 +39,182 @@
 	density = 0
 
 
-	var/obj/item/card/id/giver
+	var/obj/item/card/id/scan
 	var/list/accesses = list()
 	var/giv_name = "NOT SPECIFIED"
 	var/reason = "NOT SPECIFIED"
 	var/duration = 5
+	var/print_cooldown = 0
 
 	var/list/internal_log = list()
-	var/mode = 0  // 0 - making pass, 1 - viewing logs
+	var/mode = FALSE  // FALSE - making pass, TRUE - viewing logs
 
 /obj/machinery/computer/guestpass/attackby(obj/item/I, mob/user, params)
 	if(istype(I, /obj/item/card/id))
-		if(!giver)
+		if(!scan)
 			if(user.drop_item())
 				I.forceMove(src)
-				giver = I
+				scan = I
 				updateUsrDialog()
 		else
 			to_chat(user, "<span class='warning'>There is already ID card inside.</span>")
 		return
 	return ..()
 
 /obj/machinery/computer/guestpass/proc/get_changeable_accesses()
-	return giver.access
+	return scan.access
 
 /obj/machinery/computer/guestpass/attack_ai(mob/user)
 	return attack_hand(user)
 
 
-/obj/machinery/computer/guestpass/attack_hand(var/mob/user as mob)
-	if(..())
-		return
+/obj/machinery/computer/guestpass/attack_hand(mob/user)
+	ui_interact(user)
+
+/obj/machinery/computer/guestpass/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "GuestPass",  name, 500, 850, master_ui, state)
+		ui.open()
```

Comment:
```suggestion
		ui.open()
		ui.set_autoupdate(FALSE)
```

Reply:
Why?

Reply:
Because there is no data that is changed periodically while the window is open. All updates in this TGUI are done as part of an action. To do so periodically is wasteful.